airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/BaseS3Destination.java

```diff
@@ -47,6 +47,7 @@ public AirbyteConnectionStatus check(JsonNode config) {
       S3BaseChecks.attemptS3WriteAndDelete(storageOperations, destinationConfig, destinationConfig.getBucketName());
       S3BaseChecks.testSingleUpload(s3Client, destinationConfig.getBucketName(), destinationConfig.getBucketPath());
       S3BaseChecks.testMultipartUpload(s3Client, destinationConfig.getBucketName(), destinationConfig.getBucketPath());
+      S3BaseChecks.testCustomEndpointSecured(destinationConfig.getEndpoint());
```

Comment:
from what I understand, this will force all uses of S3 connector to use HTTPS even outside of Airbyte Cloud, which is not the intention of the issue. The intention is to force this only in Airbyte Cloud.

Reply:
Yes, I haven't pushed up the code that uses the `AdaptiveDestinationRunner` which will only set the check with custom endpoint for Cloud. I'm currently in the process of getting a local version of MinIO S3 running to test the custom endpoint doesn't fail for OSS users

Reply:
I see you pushed the changes that use AdaptiveDestinationRunner, but this part is still here

Reply:
Oh good catch, I'll update this